src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvAwaitable.cs

```diff
@@ -49,10 +49,16 @@ public UvWriteResult GetResult()
 
         public void OnCompleted(Action continuation)
         {
+            // There should never be a race between IsCompleted and OnCompleted since both operations
+            // should always be on the libuv thread
```

Comment:
I had a commit that passed the thread in but that complicated the code. This should be good enough for most cases. I also considered getting rid of the Interlocked.* operations since everything is thread safe.